shopinvader_sale_profile/models/shopinvader_backend.py

```diff
@@ -32,10 +34,45 @@ def _onchange_use_sale_profile(self):
 
     @api.constrains("use_sale_profile", "pricelist_id")
     def _check_use_sale_profile(self):
+        # TODO: does this make sense?
+        # We could set the pricelist in the onchange above
         if self.pricelist_id and self.use_sale_profile:
             raise exceptions.ValidationError(
                 _(
                     "You can not use sale profile and set a specific "
                     "pricelist at the same time."
                 )
             )
+
+    def _get_cart_pricelist(self, partner=None):
```

Comment:
we can trash part of this stuff if on sale profile default change we set:
1. pricelist_id == default prof pricelist
2. make pricelist id not editable
3. use _get_partner_pricelist (see #805)

Reply:
the best approach would be to make the sale.order deal w/ onchange_partner_id probably. To be discussed again in the context sale_channel.

Reply:
I agree, we should move on with this and look back at it when sale_channel is ok